src/uwtools/rocoto.py

```diff
@@ -182,13 +182,47 @@ def _add_task_dependency(self, e: Element, config: dict) -> None:
         :param config: Configuration data for this element.
         """
         e = SubElement(e, STR.dependency)
-        for key, block in config.items():
-            tag, _ = self._tag_name(key)
-            if tag == STR.taskdep:
-                self._set_attrs(SubElement(e, STR.taskdep), block)
+        for tag, block in config.items():
+            if tag in [STR.taskdep, STR.datadep, STR.timedep]:
+                self._add_task_dependency_operand(e, tag=tag, block=block)
+            elif tag in [STR.and_, STR.nand, STR.nor, STR.not_, STR.or_, STR.xor]:
+                self._add_task_dependency_operator(e, config={tag: block})
```

Comment:
I left this as `config` because when iterating over the `config` in the operand method, we will only want to be passing the net `block` to the recurse the method again. We do not want to be passing every `tag` again or this leads to a never ending loop. However, we need to pass the tag to add to the `dependency` tree as Christina pointed out.

Reply:
I see what you mean. Since this is recursive, the dicts may be larger in subsequent calls than they are on this first call, so make sure it can handle an entire full dict.

Reply:
@christinaholtNOAA How would the dicts get larger? Aren't we processing smaller and smaller subtrees? Or do you just mean that the number of keys at the top level might be larger?